### ISSUE TITLE
Review Continuous Integration (Nx - Affected) workflows triggers

### ISSUE THREAD

**Issue by overbit** [amplication/amplication#4429]
Currently we trigger `Continuous Integration (Nx - Affected)` workflows on:
- workflow_dispatch
- push
- pull_request_target

`nx affected` commands work out the application and library affected by a diff between a BASE and HEAD commits. https://nx.dev/concepts/affected 

In order to return rightful results, for each workflow, we will need to specify the base commit (or branch).
This is only possible in `pull_request` triggers and, with manual input, also workflow dispatch, but not in `push` since it is only aware about the commit (not branch).

As a plus, review the jobs order and raise a new issue if any work is needed

**Comment by overbit** [comment]
Blocked by https://github.com/nrwl/nx-set-shas/pull/80
